### PR TITLE
Update Enabling Report Misbehavior plugin - Closes 1532 

### LINKED ISSUE #1532: Update `How to enable the "Report Misbehavior" plugin`
### Description
- Move this page from core docs to lisk docs
- Update  `How to enable the "Report Misbehavior" plugin` to use Lisk SDK 6.0 interface.
- Update to enable the report misbehavior plugin using the new `endpoint:invoke` command.
- It should mention it is only for chains that use the PoS module.

### Additional Information
https://liskhq.github.io/lisk-docs/lisk-core/management/enabling-misbehavior-report.html

### Affected Branch
`lisk-docs/development`

### Page location
`run-blockchain/enabling-misbehavior-report`

### PR DESCRIPTION
**Pull request by TalhaMaliktz**
Closes #1532

### CODE REVIEW

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 1]
```diff
@@ -1,126 +1,219 @@
...
-:url_plugin_report_misbehavior: lisk-sdk::plugins/report-misbehavior-plugin.adoc
-
-// TODO: Update the page by uncommenting the hyperlinks once the updated pages are available. 
+:url_guides_setup: build-blockchain/create-blockchain-client.adoc
+:url_guides_config: build-blockchain/configuration.adoc
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_cli_globally: build-blockchain/create-blockchain-client.adoc#using-the-client-cli-globally
+:url_pm2_conf: run-blockchain/process-management.adoc#using-a-json-config-file
 
+The {url_plugin_report_misbehavior}[Report Misbehavior plugin^] provides automatic detection of the validator's misbehavior and sends a `reportDelegateMisbehavior` transaction to the running node.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: "delegate" sounds like old terminology, please update

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 2]
```diff
@@ -1,126 +1,219 @@
...
+
+After updating the *plugins.ts* file, build the blockchain client again:
+
+[source,bash]
+----
+npm run build
+----
 
-== PM2 configuration
+=== Registering via PM2
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: Maybe rename it, as there are also other ways to set environment variables
```suggestion
=== Enabling via environment variable
```

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 3]
```diff
@@ -1,126 +1,219 @@
...
+Add the environment variable `LISK_ENABLE_REPORT_MISBEHAVIOR_PLUGIN` to the PM2 configuration for the `hello_client`:
 
-Add the environment variable `LISK_ENABLE_REPORT_MISBEHAVIOR_PLUGIN` to the pm2 config for Lisk Core:
+NOTE: You can use the alias `*hello_client*` only if you have configured a global alias.
+For more information see, xref:{url_cli_globally}[Using the Client CLI Globally].
+Otherwise, you can set up the `*pm2.conf.json*` file as suggested in the section: xref:{url_pm2_conf}[Using a JSON config file] for PM2.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: I wouldn't include this info here, and keep it more general for other sidechains. Also, it is a little inconsistent with next section where we use `./bin/run` in the examples

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 4]
```diff
@@ -1,126 +1,219 @@
...
     "LISK_NETWORK": "testnet",
-    "LISK_CONFIG_FILE": "/home/lisk/lisk-core/custom-config.json",
+    "LISK_CONFIG_FILE": "~/.lisk/hello_client/config/custom-config.json",
     "LISK_ENABLE_REPORT_MISBEHAVIOR_PLUGIN": true
   }
 }
 ----
 
-== Restart the node
+== Starting the node
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc:140]: I would restructure this section to already be included separately in the above examples

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 5]
```diff
@@ -1,126 +1,219 @@
...
+
 
-== Authorization
+== Authorizing an account
 
-The "Report Misbehavior" plugin provides a dedicated action to enable and disable the plugin on the node.
+=== Choose the sender account
 
-For example, use the `lisk-client` package and write a small script that invokes the `reportMisbehavior:authorize` action on the node.
+The plugin uses a specified sender account to sign the `reportDelegateMisbehavior` transactions.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: outdated terminology

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc (hunk 3 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: Actually, in most cases users want to enable it on Lisk Core, so maybe we should include the respective examples. Either as tabs, starting with Lisk Core examples followed by SDK, or replace the existing examples with Lisk Core

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 6]
```diff
@@ -1,126 +1,219 @@
...
+====
+On this page, you'll learn:
 
-To enable the plugin on your node, perform the following actions:
+* [x] Registration of the Report Misbehavior plugin to a blockchain client.
+* [x] Authorizing an account to report misbehavior on the network.
+====
 
-== Choose sender account
+For this guide, the Report Misbehavior plugin is registered to the xref:{url_guides_setup}[Hello World application], though the steps mentioned below can be applied to any Lisk-based blockchain client.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: Would also explain for Lisk Core, see comment below

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 7]
```diff
@@ -1,126 +1,219 @@
...
-
-// TODO: Update the page by uncommenting the hyperlinks once the updated pages are available. 
+:url_guides_setup: build-blockchain/create-blockchain-client.adoc
+:url_guides_config: build-blockchain/configuration.adoc
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_cli_globally: build-blockchain/create-blockchain-client.adoc#using-the-client-cli-globally
+:url_pm2_conf: run-blockchain/process-management.adoc#using-a-json-config-file
 
+The {url_plugin_report_misbehavior}[Report Misbehavior plugin^] provides automatic detection of the validator's misbehavior and sends a `reportDelegateMisbehavior` transaction to the running node.
+The plugin is designed to work with PoS based blockchains only.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: Maybe you can add a little more info here what misbehavior exactly will be reported, and maybe refer to the corresponding LIP as well, in a footnote for example.

**Review by Tschakki — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 8]
```diff
@@ -1,60 +1,109 @@
...
+* [x] Authorizing an account to report misbehavior on the network.
+====
 
-== Choose sender account
+Using this guide, the Report Misbehavior plugin can be registered to a xref:{url_core_index}[Lisk Core] or a xref:{url_guides_setup}[Hello World application] node, though the steps mentioned below can be applied to any Lisk-based blockchain client.
 
-The plugin will use a specified sender account to sign the `reportDelegateMisbehavior` transactions.
-To automatically sign the transaction, the encrypted passphrase of the account must be added to the plugin configuration.
+== Plugin registration
+With the latest version of Lisk Core v4 and Lisk SDK v6, the default plugins come pre-installed after an client is installed or initiated, respectively.
```
Comment (Incede) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]:
```suggestion
With the latest version of Lisk Core v4 and Lisk SDK v6, the default plugins come pre-installed after a client is installed or initiated, respectively.
```

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 9]
```diff
@@ -1,60 +1,109 @@
...
+[IMPORTANT]
+====
+It is essential to enable WS API and configure the `allowedMethods` property before registering the `ReportMisbehaviorPlugin`.
+For more information, see the configurations described in the xref:{url_guides_config_hello}[configuration guide].
+====
 
-.custom-config.json
-[source,json]
+=== Registering via source
+For a *sidechain*, It is possible to register the plugin through the `plugins.ts` file.
```
Comment (Incede) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: question for my understanding - why is it not possible to do the same for mainchain ?

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 10]
```diff
@@ -68,59 +117,194 @@ Add the environment variable `LISK_ENABLE_REPORT_MISBEHAVIOR_PLUGIN` to the pm2
...
+
+== Starting the node
+Depending on how you chose to register the plugin, start the blockchain client, accordingly.
+
+[tabs]
+=====
+Registered via built-in flag::
++
+--
+It is possible to enable the Report Misbehavior plugin without updating the configuration, the `plugins.ts` file, or using PM2.
```
Comment (Incede) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]:
```suggestion
It is also possible to enable the Report Misbehavior plugin without updating the configuration, the `plugins.ts` file, or using PM2.
```

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 11]
```diff
@@ -68,59 +117,194 @@ Add the environment variable `LISK_ENABLE_REPORT_MISBEHAVIOR_PLUGIN` to the pm2
...
-		password: "myPassword",
-        enable: true
-    }).then(res => {
-		console.log(res);
-		process.exit(0);
-	});
-});
+=== Plugin configuration
+
+Add the encrypted private key retrieved above under `plugins.reportMisbehavior.encryptedPrivateKey` in the node configuration:
```
Comment (Incede) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]:
```suggestion
Add the encrypted private key retrieved above under `plugins.reportMisbehavior.encryptedPrivateKey` to the node configuration:
```

**Review by Incede — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor points

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc (hunk 9 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: `lisk-core` comes pre-configured with the default plugins. There is no `plugins.ts` file in `lisk-core` which can be edited to register a plugin via source.

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 12]
```diff
@@ -1,60 +1,109 @@
...
+:url_guides_config_hello: {url_guides_config}#example-configuration-for-the-hello-world-client
+:url_cli_globally: build-blockchain/create-blockchain-client.adoc#using-the-client-cli-globally
+:url_pm2_conf: run-blockchain/process-management.adoc#using-a-json-config-file
+:url_core_index: v4@lisk-core::index.adoc
 
-// TODO: Update the page by uncommenting the hyperlinks once the updated pages are available. 
+:fn_lip24: footnote:witness[The LIP: {url_consensus_bft_punishment}[Punish BFT violations^] provides the details regarding the implications of {url_rep_mis_transaction}[Report Misbehavior^] transaction.]
 
+The {url_plugin_report_misbehavior}[Report Misbehavior plugin^] provides automatic detection of a validator's misbehavior and sends the `reportMisbehavior` transaction to the running node.
+This transaction contains the information necessary to prove that the validator who signed the {url_blocks_header}[block headers^] has violated the {url_consensus_bft}[Lisk-BFT consensus protocol^] and is deemed to be punished{fn_lip24} once proven guilty.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: not sure why we are referring here to the block header specifically? I would just say "the validator who generated the block", to keep it simple.

Comment (Tschakki) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: A ToDo should be added here to update the link to BFT LIP with link to docs BFT page, once the page is created

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 13]
```diff
@@ -1,60 +1,109 @@
...
 :sectnums:
+:experimental:
+
 // External URLs
+:url_plugin_report_misbehavior: https://lisk.com/documentation/lisk-sdk/v6/references/typedoc/modules/_liskhq_lisk_framework_report_misbehavior_plugin.html
+:url_plugin_report_misbehavior_config: https://lisk.com/documentation/lisk-sdk/v6/references/typedoc/modules/_liskhq_lisk_framework_report_misbehavior_plugin.html#$config-options
+:url_blocks_header: https://github.com/LiskHQ/lips/blob/main/proposals/lip-0055.md#block-header
+:url_consensus_bft: https://github.com/LiskHQ/lips/blob/1e27a5ba99a55d71e6d83f44b6e21f18dda98741/proposals/lip-0014.md
+:url_consensus_bft_punishment: https://github.com/LiskHQ/lips/blob/1e27a5ba99a55d71e6d83f44b6e21f18dda98741/proposals/lip-0024.md
+:url_rep_mis_transaction: https://github.com/LiskHQ/lips/blob/1e27a5ba99a55d71e6d83f44b6e21f18dda98741/proposals/lip-0057.md#report-misbehavior
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]:
```suggestion
:url_consensus_bft: https://github.com/LiskHQ/lips/blob/main/proposals/lip-0014.md
:url_consensus_bft_punishment: https://github.com/LiskHQ/lips/blob/main/proposals/lip-0024.md
:url_rep_mis_transaction: https://github.com/LiskHQ/lips/blob/main/proposals/lip-0057.md#report-misbehavior
```

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 14]
```diff
@@ -1,60 +1,109 @@
...
+====
+On this page, you'll learn:
 
-To enable the plugin on your node, perform the following actions:
+* [x] Registration of the Report Misbehavior plugin to a blockchain client.
+* [x] Authorizing an account to report misbehavior on the network.
+====
 
-== Choose sender account
+Using this guide, the Report Misbehavior plugin can be registered to a xref:{url_core_index}[Lisk Core] or a xref:{url_guides_setup}[Hello World application] node, though the steps mentioned below can be applied to any Lisk-based blockchain client.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]:
```suggestion
Using this guide, the Report Misbehavior plugin can be registered to a xref:{url_core_index}[Lisk Core] node or any Lisk-based blockchain client.
```

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 15]
```diff
@@ -1,60 +1,109 @@
...
 
-The plugin will use a specified sender account to sign the `reportDelegateMisbehavior` transactions.
-To automatically sign the transaction, the encrypted passphrase of the account must be added to the plugin configuration.
+== Plugin registration
+With the latest version of Lisk Core v4 and Lisk SDK v6, the default plugins come pre-installed after a client is installed or initiated, respectively.
 
-NOTE: Please note that the respective account should have a high enough account balance to be able to send a misbehavior report transaction.
+For a sidechain, you can check the `package.json` file of the blockchain client to ensure the installation.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: This is also valid for Lisk Core.

```suggestion
You can check the `package.json` file of the blockchain client to ensure the installation.
```

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc [hunk 16]
```diff
@@ -1,60 +1,109 @@
...
 ----
 
-== Plugin configuration
+There are various ways to enable the Report Misbehavior plugin.
+You can opt for any of the below-mentioned to register the plugin to the blockchain client.
 
-Add the encrypted passphrase to the configuration under `plugins.reportMisbehavior.encryptedPassphrase` in the node configuration:
+[IMPORTANT]
+====
+It is essential to enable WS API and configure the `allowedMethods` property before registering the `ReportMisbehaviorPlugin`.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc:70]: How exactly should it be configured for the plugin? Please elaborate.

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc (hunk 9 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]:
```suggestion
For blockchain clients that were initialized with Lisk Commander, it is possible to register the plugin through the `plugins.ts` file.
```

**Review by Tschakki — CHANGES_REQUESTED** [review]
Looking good already, just a few minor comments

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc (hunk 12 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: If you run the command `reportMisbehavior`, it expects two header values to be passed to it, which is why I have mentioned the headers to be specific. To be exact, signing a false block is a misbehavior that should be reported.

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc (hunk 16 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc:70]: Ummmm! The details are mentioned in the URL referenced in the next line.

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc (hunk 12 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]: In this case, I would rephrase it to make it clearer, and also, please replace link to block header with internal link to https://liskhq.github.io/lisk-docs/beta/understand-blockchain/blocks-txs.html#block-header


```suggestion
This transaction contains two xref:{}[block headers] necessary to prove that the validator who generated the blocks has violated the {url_consensus_bft}[Lisk-BFT consensus protocol^] and is deemed to be punished{fn_lip24} once proven guilty.
```

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc (hunk 16 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc:70]: In the URL we explain to set allowedMethods to "*", but here for the guide we should mention the specific methods that should be enabled for this plugin to work correctly.

File: docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc (hunk 12 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/run-blockchain/enabling-misbehavior-report.adoc]:
```suggestion
This transaction contains two xref:{url_blocks_header}[block headers] that are necessary to prove that the validator who generated the blocks has violated the {url_consensus_bft}[Lisk-BFT consensus protocol^] and is deemed to be punished{fn_lip24} once proven guilty.
```

**Review by Tschakki — APPROVED** [review]
LGTM

**Review by Incede — APPROVED** [review]
